File: Find element occuring once when all other are present thrice.cpp
class Solution {
  public:
    int singleElement(int arr[] ,int N) {
        // code here
        unordered_map<int, int>mpp;
        for(int i=0; i<N; i++){
            mpp[arr[i]]++;
        }
        for(auto &it:mpp){
            if(it.second==1) return it.first;
        }
        return -1;
    }
};

File: Find element occuring once when all other are present thrice.java
class Solution {
    static int singleElement(int[] arr , int N) {
        // code here
       HashMap<Integer, Integer> hm = new HashMap<>();
       for(int i=0;i<N; i++){
           if(!hm.containsKey(arr[i])){
               hm.put(arr[i],1);
           }    else{
               Integer a = hm.get(arr[i]);
               hm.put(arr[i],a.intValue() + 1);
           }
           
       }
        for(Map.Entry<Integer, Integer> entry : hm.entrySet()){
            if(entry.getValue() == 1){
                return entry.getKey();
            }
        }
            return -1;
   }
}
